File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  // Trim whitespace from the message
  const trimmedMessage = message.trim();
  
  // Check for silence (empty or only whitespace)
  if (trimmedMessage === '') {
    return 'Fine. Be that way!';
  }
  
  // Check if the message is yelling (contains letters and all letters are uppercase)
  const hasLetters = /[a-zA-Z]/.test(trimmedMessage);
  const isYelling = hasLetters && trimmedMessage === trimmedMessage.toUpperCase();
  
  // Check if the message is a question (ends with '?')
  const isQuestion = trimmedMessage.endsWith('?');
  
  // Determine the response based on the conditions
  if (isYelling && isQuestion) {
    return "Calm down, I know what I'm doing!";
  } else if (isYelling) {
    return 'Whoa, chill out!';
  } else if (isQuestion) {
    return 'Sure.';
  } else {
    return 'Whatever.';
  }
}

File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {

    String hey(String input) {
        String trimmed = input.trim();
        
        // Check for silence first
        if (trimmed.isEmpty()) {
            return "Fine. Be that way!";
        }
        
        boolean isQuestion = trimmed.endsWith("?");
        boolean isYelling = isYelling(trimmed);
        
        if (isYelling && isQuestion) {
            return "Calm down, I know what I'm doing!";
        }
        if (isYelling) {
            return "Whoa, chill out!";
        }
        if (isQuestion) {
            return "Sure.";
        }
        
        return "Whatever.";
    }
    
    // Helper: determines if text is "yelling" (contains letters, all uppercase)
    private boolean isYelling(String text) {
        boolean hasLetter = false;
        for (int i = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            if (Character.isLetter(c)) {
                hasLetter = true;
                if (Character.isLowerCase(c)) {
                    return false; // Found a lowercase letter -> not yelling
                }
            }
        }
        return hasLetter; // Must contain at least one letter to be yelling
    }
}